From adbbe88a19701460de32e30ce4af355b905d7a51 Mon Sep 17 00:00:00 2001
From: Tobias <ruck.tobias@gmail.com>
Date: Mon, 27 Jun 2022 01:50:57 +0200
Subject: [PATCH 166/196] Unify coinbase addresses of all 3 nets (#257)

Currently, testnet uses different coinbase addresses than mainnet. This change makes them use all the same addresses, and also removes the need for keeping 3 seperate files.
---
 src/chainparams.cpp                           | 13 +--
 .../{addresses_mainnet.h => addresses.h}      |  2 -
 src/consensus/addresses_regtest.h             | 94 -------------------
 src/consensus/addresses_testnet.h             | 94 -------------------
 .../logos_feature_minerfund_activation.py     |  2 +-
 5 files changed, 5 insertions(+), 200 deletions(-)
 rename src/consensus/{addresses_mainnet.h => addresses.h} (98%)
 delete mode 100644 src/consensus/addresses_regtest.h
 delete mode 100644 src/consensus/addresses_testnet.h

diff --git a/src/chainparams.cpp b/src/chainparams.cpp
index 90e387d0b..afeaef625 100644
--- a/src/chainparams.cpp
+++ b/src/chainparams.cpp
@@ -17,9 +17,7 @@
 #include <util/strencodings.h>
 #include <util/system.h>
 
-#include <consensus/addresses_mainnet.h>
-#include <consensus/addresses_regtest.h>
-#include <consensus/addresses_testnet.h>
+#include <consensus/addresses.h>
 
 #include <cassert>
 
@@ -91,8 +89,7 @@ public:
 
         // The miner fund is enabled by default on mainnet.
         consensus.enableMinerFund = ENABLE_MINER_FUND;
-        consensus.coinbasePayoutAddresses =
-            RewardAddresses::MainNet::AddressSets;
+        consensus.coinbasePayoutAddresses = RewardAddresses::AddressSets;
         assert(consensus.coinbasePayoutAddresses.genesis.size() == 13);
         // Mainnet rewards based on difficulty.
         consensus.enableDifficultyBasedSubsidy = true;
@@ -220,8 +217,7 @@ public:
 
         // The miner fund is enabled by default on testnet.
         consensus.enableMinerFund = ENABLE_MINER_FUND;
-        consensus.coinbasePayoutAddresses =
-            RewardAddresses::TestNet::AddressSets;
+        consensus.coinbasePayoutAddresses = RewardAddresses::AddressSets;
         assert(consensus.coinbasePayoutAddresses.genesis.size() == 13);
 
         // Testnet rewards based on difficulty.
@@ -327,8 +323,7 @@ public:
 
         // The miner fund is disabled by default on regnet.
         consensus.enableMinerFund = false;
-        consensus.coinbasePayoutAddresses =
-            RewardAddresses::RegTest::AddressSets;
+        consensus.coinbasePayoutAddresses = RewardAddresses::AddressSets;
         assert(consensus.coinbasePayoutAddresses.genesis.size() == 13);
         // Regtest rewards a constant amount independent of difficulty.
         consensus.enableDifficultyBasedSubsidy = false;
diff --git a/src/consensus/addresses_mainnet.h b/src/consensus/addresses.h
similarity index 98%
rename from src/consensus/addresses_mainnet.h
rename to src/consensus/addresses.h
index ad0af03d7..50d24a511 100644
--- a/src/consensus/addresses_mainnet.h
+++ b/src/consensus/addresses.h
@@ -4,7 +4,6 @@
 #include <consensus/params.h>
 
 namespace RewardAddresses {
-namespace MainNet {
     Consensus::CoinbaseAddresses AddressSets = {
         .genesis = {
             // Foundation
@@ -90,5 +89,4 @@ namespace MainNet {
             // Lotus Citizen 4
             "lotus_16PSJNEqzEoLhBigDN8Hf93FzERrPQKJXaYdzRCNT",
         }};
-} // namespace MainNet
 } // namespace RewardAddresses
diff --git a/src/consensus/addresses_regtest.h b/src/consensus/addresses_regtest.h
deleted file mode 100644
index f5346d8ea..000000000
--- a/src/consensus/addresses_regtest.h
+++ /dev/null
@@ -1,94 +0,0 @@
-#include <string>
-#include <vector>
-
-#include <consensus/params.h>
-
-namespace RewardAddresses {
-namespace RegTest {
-    Consensus::CoinbaseAddresses AddressSets = {
-        .genesis = {
-            // Foundation
-            "pzmv0yp3kuwcd2cdv9lpu8nsdmzwud9s0upp4rxwc9",
-            // Bitcoin ABC
-            "qzu2u8z8ala43ae0009gr8l8lsjjl8599cdwdr9mxq",
-            // Tobias
-            "qz6shp4gj0vqe83wuu43n9sjxa9hknqumq0xdtwprp",
-            // Saipan Institute
-            "qrd8dgcmvasdewg2535uzkt9mfhgqztwg504dhsfxc",
-            // Stamp
-            "qqfjt5kchfhgclvelandy9fsj97vwdpf6g7sfclqsh",
-            // Nghia
-            "qp4pwxy34w2yxqst6f64aauudev7l3vjv5xg6dhc7x",
-            // Remark
-            "qqv73kxsak4ka3plqjm9d0lh9tmc6cllv5m9np8ndy",
-            // Bitcoin Not Bombs
-            "qrryjt2wgnwdqpg7vz52m440q2e09ydj4gupmhu3g6",
-            // Distro Bot
-            "qz66atlvnu5hyygvf340j5y283q7r5788vkcu6qff3",
-            // Distro Bot
-            "qp7z3253hyl6lzhwpfjjpg9g8apdh39ytvlwfx57mm",
-            // Evangelism
-            "qzccavyvf9uwwdyqwsa3txqxr5708rss4qqg8p6an4",
-            // Distro Bot
-            "qp7s3y735fut4vn70tvjakyt6lww4lvr55ljvaxpyn",
-            // Distro Bot
-            "qp9cd8u62hzhqq7lz79aeqq3ssgfmyz03vc5c4g0ct",
-        },
-        .exodus = {
-            // Aleph 0
-            "lotus_16PSJNf1EDEfGvaYzaXJCJZrXH4pgiTo7kyW61iGi",
-            // Maintainance
-            "lotus_16PSJNZhPZSyLeQaiEdy3a3gZdwQ2yAPqUm7WHRzN",
-            // Stamp
-            "lotus_16PSJHhULzYgupzGnvDysUFLxYxBVwx92JjXSyjNL",
-            // VN
-            "lotus_16PSJLMtCuFnyMb3rz4zbbhAtJgppp8GiY5NpGGev",
-            // Explorer
-            "lotus_16PSJJQGXSV1uMG2JHERQpVL3pVH7nwqu2YSiX96t",
-            // Distro
-            "lotus_16PSJHEXKQEL5M28Rc8rvYntymdKtPmVXJDEtbFLi",
-            // Evangelism
-            "lotus_16PSJNnGguQxEhHHW569L3g3MfhCNsS1Q6AWrJ8AH",
-            // Lotusia Citizen 1
-            "lotus_16PSJP5wAt1F45LycobZssN8Jk3TaR6QL9AXcRRty",
-            // Marketing
-            "lotus_16PSJHej3n56z43NC4PgYAAubk8opBXt7JXXfHjwc",
-            // Lotusia Citizen 2
-            "lotus_16PSJJSRTKjXKkvMtV6NX8SVEbTBwpjCWxQP7bHdi",
-            // Bots
-            "lotus_16PSJHuUsbDo5qZUjygQgtJLMsZgcpSB6wD459F6J",
-            // Bot
-            "lotus_16PSJJXVKnYb8xtUhHyBDSdd8P2xx6suUM7TaWBNx",
-            // Lotusia Citizen 3
-            "lotus_16PSJKHmQ2Lm8BNtnvPi1jC4rxJpCo8VqvJecj5Hb",
-        },
-        .leviticus = {
-            // Aleph 0
-            "lotus_16PSJQTQMPHoT5AuGLm2UHcmbKLEFR43gHJChzTUy",
-            // Maintainance
-            "lotus_16PSJJdodJ4VMACYzBd1ey1MiPaaT5XSWVnoxQvci",
-            // Stamp
-            "lotus_16PSJHhULzYgupzGnvDysUFLxYxBVwx92JjXSyjNL",
-            // VN
-            "lotus_16PSJLMtCuFnyMb3rz4zbbhAtJgppp8GiY5NpGGev",
-            // Explorer 2
-            "lotus_16PSJMBaU1g7jvzCrXUCK5gxP6GHnSxEBYR3geqtX",
-            // Sales
-            "lotus_16PSJQ3Wby3uP6XkPtKTULTFzMXU9JJXfr1UrzTZJ",
-            // Radical Evangelism
-            "lotus_16PSJJjTw855ea3eJqEPTsuG6PkmLJw5eW4WVUjS7",
-            // Buddhism
-            "lotus_16PSJQEzUwA2bpbkzvAiDtAviX1Ep5x9x5UhdcBV6",
-            // VN 2
-            "lotus_16PSJNgypXoCzC2yQqz7wTGFCSL25qv4XZhxvQ28A",
-            // Ministry Of Truth
-            "lotus_16PSJKEP97unf4XFVR6zzS3j2s3qJp8omSJkyyDWH",
-            // Podcasters
-            "lotus_16PSJMwWGF6jNTbF8WgMuM49DZNmsWgY3b5e4itvd",
-            // Bots
-            "lotus_16PSJHuUsbDo5qZUjygQgtJLMsZgcpSB6wD459F6J",
-            // Lotus Citizen 4
-            "lotus_16PSJNEqzEoLhBigDN8Hf93FzERrPQKJXaYdzRCNT",
-        }};
-} // namespace RegTest
-} // namespace RewardAddresses
diff --git a/src/consensus/addresses_testnet.h b/src/consensus/addresses_testnet.h
deleted file mode 100644
index 848366ffa..000000000
--- a/src/consensus/addresses_testnet.h
+++ /dev/null
@@ -1,94 +0,0 @@
-#include <string>
-#include <vector>
-
-#include <consensus/params.h>
-
-namespace RewardAddresses {
-namespace TestNet {
-    Consensus::CoinbaseAddresses AddressSets = {
-        .genesis = {
-            // Foundation
-            "pzmv0yp3kuwcd2cdv9lpu8nsdmzwud9s0upp4rxwc9",
-            // Bitcoin ABC
-            "ppppwzhcy6kyyhkjsw0s7vvyzxz73vnwzynkaqjssk",
-            // Be.Cash
-            "qpwm9atcqdmr2dutnpp5h0t0hmy0y6k595qg0z5egm",
-            // Saipan Institute
-            "qrd8dgcmvasdewg2535uzkt9mfhgqztwg504dhsfxc",
-            // Stamp
-            "qqfjt5kchfhgclvelandy9fsj97vwdpf6g7sfclqsh",
-            // Nghia
-            "qp4pwxy34w2yxqst6f64aauudev7l3vjv5xg6dhc7x",
-            // Remark
-            "qqv73kxsak4ka3plqjm9d0lh9tmc6cllv5m9np8ndy",
-            // Bitcoin Not Bombs
-            "qrryjt2wgnwdqpg7vz52m440q2e09ydj4gupmhu3g6",
-            // Distro Bot
-            "qz66atlvnu5hyygvf340j5y283q7r5788vkcu6qff3",
-            // Distro Bot
-            "qp7z3253hyl6lzhwpfjjpg9g8apdh39ytvlwfx57mm",
-            // Evangelism
-            "qzccavyvf9uwwdyqwsa3txqxr5708rss4qqg8p6an4",
-            // Distro Bot
-            "qp7s3y735fut4vn70tvjakyt6lww4lvr55ljvaxpyn",
-            // Koush
-            "qrgqdz7ww5v8yqgu9ns4av5m4t4w7jxgyyxc05z9gu",
-        },
-        .exodus = {
-            // Aleph 0
-            "lotus_16PSJNf1EDEfGvaYzaXJCJZrXH4pgiTo7kyW61iGi",
-            // Maintainance
-            "lotus_16PSJNZhPZSyLeQaiEdy3a3gZdwQ2yAPqUm7WHRzN",
-            // Stamp
-            "lotus_16PSJHhULzYgupzGnvDysUFLxYxBVwx92JjXSyjNL",
-            // VN
-            "lotus_16PSJLMtCuFnyMb3rz4zbbhAtJgppp8GiY5NpGGev",
-            // Explorer
-            "lotus_16PSJJQGXSV1uMG2JHERQpVL3pVH7nwqu2YSiX96t",
-            // Distro
-            "lotus_16PSJHEXKQEL5M28Rc8rvYntymdKtPmVXJDEtbFLi",
-            // Evangelism
-            "lotus_16PSJNnGguQxEhHHW569L3g3MfhCNsS1Q6AWrJ8AH",
-            // Lotusia Citizen 1
-            "lotus_16PSJP5wAt1F45LycobZssN8Jk3TaR6QL9AXcRRty",
-            // Marketing
-            "lotus_16PSJHej3n56z43NC4PgYAAubk8opBXt7JXXfHjwc",
-            // Lotusia Citizen 2
-            "lotus_16PSJJSRTKjXKkvMtV6NX8SVEbTBwpjCWxQP7bHdi",
-            // Bots
-            "lotus_16PSJHuUsbDo5qZUjygQgtJLMsZgcpSB6wD459F6J",
-            // Bot
-            "lotus_16PSJJXVKnYb8xtUhHyBDSdd8P2xx6suUM7TaWBNx",
-            // Lotusia Citizen 3
-            "lotus_16PSJKHmQ2Lm8BNtnvPi1jC4rxJpCo8VqvJecj5Hb",
-        },
-        .leviticus = {
-            // Aleph 0
-            "lotus_16PSJQTQMPHoT5AuGLm2UHcmbKLEFR43gHJChzTUy",
-            // Maintainance
-            "lotus_16PSJJdodJ4VMACYzBd1ey1MiPaaT5XSWVnoxQvci",
-            // Stamp
-            "lotus_16PSJHhULzYgupzGnvDysUFLxYxBVwx92JjXSyjNL",
-            // VN
-            "lotus_16PSJLMtCuFnyMb3rz4zbbhAtJgppp8GiY5NpGGev",
-            // Explorer 2
-            "lotus_16PSJMBaU1g7jvzCrXUCK5gxP6GHnSxEBYR3geqtX",
-            // Sales
-            "lotus_16PSJQ3Wby3uP6XkPtKTULTFzMXU9JJXfr1UrzTZJ",
-            // Radical Evangelism
-            "lotus_16PSJJjTw855ea3eJqEPTsuG6PkmLJw5eW4WVUjS7",
-            // Buddhism
-            "lotus_16PSJQEzUwA2bpbkzvAiDtAviX1Ep5x9x5UhdcBV6",
-            // VN 2
-            "lotus_16PSJNgypXoCzC2yQqz7wTGFCSL25qv4XZhxvQ28A",
-            // Ministry Of Truth
-            "lotus_16PSJKEP97unf4XFVR6zzS3j2s3qJp8omSJkyyDWH",
-            // Podcasters
-            "lotus_16PSJMwWGF6jNTbF8WgMuM49DZNmsWgY3b5e4itvd",
-            // Bots
-            "lotus_16PSJHuUsbDo5qZUjygQgtJLMsZgcpSB6wD459F6J",
-            // Lotus Citizen 4
-            "lotus_16PSJNEqzEoLhBigDN8Hf93FzERrPQKJXaYdzRCNT",
-        }};
-} // namespace TestNet
-} // namespace RewardAddresses
diff --git a/test/functional/logos_feature_minerfund_activation.py b/test/functional/logos_feature_minerfund_activation.py
index 756476bbd..f2ddaf4fe 100755
--- a/test/functional/logos_feature_minerfund_activation.py
+++ b/test/functional/logos_feature_minerfund_activation.py
@@ -28,7 +28,7 @@ from test_framework.util import assert_equal
 EXODUS_ACTIVATION_TIME = 2000000000
 LEVITICUS_ACTIVATION_TIME = 2010000000
 
-# see consensus/addresses_mainnet.h
+# see consensus/addresses.h
 GENESIS_SCRIPTS = [
     "a914b6c79031b71d86ab0d617e1e1e706ec4ee34b07f87",
     "76a914b8ae1c47effb58f72f7bca819fe7fc252f9e852e88ac",
-- 
2.39.5 (Apple Git-154)

